docs/registry/README.md

```diff
@@ -0,0 +1,299 @@
+# Registry Proposal
+
+## Objective
+
+Design an **initial** version of the Registry that can support discoverability of 
+the different event types that can be consumed from the eventing mesh.
```

Comment:
I'll try but I doubt I'll be able to make the call today... but we can more added here to explain why this is needed and why going directly to the event producer for this info (like people do today) isn't sufficient? As a user of this system, unless the Registry contains all possible event types produced (and that I can see) then it doesn't provide me much value over going directly to the producer. For example, only have the list of events "seen" doesn't seem very interesting since it's almost random. If the events in the registry do not align (naming-wise) with that I see from the producer, then it's less interesting to me because I'm now locked-in. 

So far this doc explains the design but not the rationale behind why we're doing it and what the benefit is to the user over what they do today.